packages/yourloops/components/patient-data.tsx

```diff
@@ -60,35 +61,47 @@ function PatientDataPageError({ msg }: PatientDataPageErrorProps): JSX.Element {
 
 function PatientDataPage(): JSX.Element | null {
   const paramHook = useParams();
+  const authHook = useAuth();
   const teamHook = useTeam();
   const dataHook = useData();
 
   const [patient, setPatient] = React.useState<Readonly<TeamUser> | null>(null);
   const [error, setError] = React.useState<string | null>(null);
 
   const { blipApi } = dataHook;
-  const { patientId } = paramHook as PatientDataParam;
+  const { patientId: paramPatientId } = paramHook as PatientDataParam;
+  const userId = authHook.user?.userid;
+
+  const initialized = authHook.initialized() && teamHook.initialized && blipApi !== null;
 
   React.useEffect(() => {
-    if (blipApi === null) {
+    if (!initialized) {
       return;
     }
 
-    if (typeof patientId === "undefined") {
+    let patientId: string | null = null;
+    if (typeof paramPatientId === "string") {
```

Comment:
```suggestion
    if (paramPatientId) {
```
I guess you want to test that paramPatientId is not undefined, to me this way of writing is a bit confusing for the reader.

Reply:
In fact, I think we can just write it like this:
`const  patientId: string | null = paramPatientId ?? userId`

Reply:
Resolved in my own way 😉